### PR TITLE
VP-2761 Fix IndexChangesJob does not index new Imported productds.

### PR DESCRIPTION
**Pull request by trueboroda**
When the event base indexation is turned off.

### CODE REVIEW

File: src/VirtoCommerce.CatalogModule.Data/Search/Indexing/ProductDocumentChangesProvider.cs [hunk 1]
```diff
@@ -39,15 +39,24 @@ public virtual async Task<long> GetTotalChangesCountAsync(DateTime? startDate, D
             }
             else
             {
+                // Get added and modified products count
+                using (var repository = _catalogRepositoryFactory())
+                {
+                    result = await repository.Items.CountAsync(i => i.ParentId == null && ((i.ModifiedDate >= startDate && i.ModifiedDate <= endDate) || (i.CreatedDate >= startDate && i.CreatedDate <= endDate)));
```
Comment (yecli) [inline src/VirtoCommerce.CatalogModule.Data/Search/Indexing/ProductDocumentChangesProvider.cs:45]: IS this query working as expected if startDate or endDate = null?

Comment (trueboroda) [reply src/VirtoCommerce.CatalogModule.Data/Search/Indexing/ProductDocumentChangesProvider.cs:45]: It is an excepted case. In that case, the system should throw an argument exception.  It was done.

Comment (yecli) [reply src/VirtoCommerce.CatalogModule.Data/Search/Indexing/ProductDocumentChangesProvider.cs:45]: The same validation should be done in GetTotalChangesCountAsync, isn't it?

File: src/VirtoCommerce.CatalogModule.Data/Search/Indexing/ProductDocumentChangesProvider.cs [hunk 2]
```diff
@@ -56,56 +52,112 @@ public virtual async Task<long> GetTotalChangesCountAsync(DateTime? startDate, D
...
+
+                    var modifiedProductIndexDocumentChanges = await GetModifiedProductIndexDocumentChanges(startDate, endDate, skip, take, repository);
+                    result.AddRange(modifiedProductIndexDocumentChanges);
+                }
             }
 
             return result;
         }
+
+        private async Task<ChangeLogSearchResult> SearchDeleteOperationIntoLog(DateTime? startDate, DateTime? endDate, long skip, long take)
```
Comment (tatarincev) [inline src/VirtoCommerce.CatalogModule.Data/Search/Indexing/ProductDocumentChangesProvider.cs:93]: SearchDeleteOperationIntoLog -> SearchDeleteOperationInLog

**Pull request merged**